### PR TITLE
removed gbmachine object from interface

### PR DESCRIPTION
**Pull request by gf712**

### CODE REVIEW

File: src/shogun/machine/StochasticGBMachine.cpp [hunk 1]
```diff
@@ -401,11 +401,11 @@ void CStochasticGBMachine::init()
...
-	SG_ADD(&m_learning_rate,"m_learning_rate","learning rate");
-	SG_ADD((CSGObject**)&m_weak_learners,"m_weak_learners","array of weak learners");
-	SG_ADD((CSGObject**)&m_gamma,"m_gamma","array of learner weights");
+	SG_ADD(&m_machine, kMachine, "machine");
+	SG_ADD(&m_loss, kLoss, "loss function");
+	SG_ADD(&m_num_iter, kNumIterations, "number of iterations");
+	SG_ADD(&m_subset_frac, kSubsetFrac, "subset fraction");
+	SG_ADD(&m_learning_rate, kLearningRate, "learning rate");
+	SG_ADD(&m_weak_learners, kWeakLearners, "array of weak learners");
+	SG_ADD((CSGObject**)&m_gamma, kGamma, "array of learner weights");
```
Comment (gf712) [inline src/shogun/machine/StochasticGBMachine.cpp:410]: registering m_gamma as CDynamicArray<float64_t>* causes issues with cloning

Comment (vigsterkr) [reply src/shogun/machine/StochasticGBMachine.cpp:410]: mmm that should be std::vector<float64_t> but yeah we'll take care of that laterz

Comment (gf712) [reply src/shogun/machine/StochasticGBMachine.cpp:410]: yea, just wasn't sure if everything was ready to start registering std::vector

Comment (karlnapf) [reply src/shogun/machine/StochasticGBMachine.cpp:410]: I think you can :)

**Pull request merged**